src/libtriton/arch/x86/x86Semantics.cpp

```diff
@@ -7530,6 +7570,38 @@ namespace triton {
       }
 
 
+      void x86Semantics::movss_s(triton::arch::Instruction& inst) {
+        auto& dst = inst.operands[0];
+        auto& src = inst.operands[1];
+
+        /* Create symbolic operands */
+        auto op = this->symbolicEngine->getOperandAst(inst, src);
+
+        /* Create the semantics */
+        auto node = op;
+        if (src.getType() == OP_REG) {
+          node = this->astCtxt->extract(triton::bitsize::dword-1, 0, node);
+          if (dst.getType() == OP_REG) {
+            auto op1 = this->symbolicEngine->getOperandAst(inst, dst);
+            auto upper = this->astCtxt->extract(triton::bitsize::dqword-1, triton::bitsize::dword, op1);
+            node = this->astCtxt->concat(upper, node);
+          }
+        }
+        else {
+          node = this->astCtxt->concat(this->astCtxt->bv(0, triton::bitsize::dqword - triton::bitsize::dword), node);
```

Comment:
No need to concat with zero.

SymbolicEngine::createSymbolicRegisterExpression -> SymbolicEngine::insertSubRegisterInParent will perform zero extension.

Reply:
Removed.